### PR TITLE
add pybind to get eigenvalues of a matrix

### PR DESCRIPTION
**Pull request by skkwan**
This PR adds a function to get the eigenvalues of a matrix. For example:

```python
import tac_hep_coding_jam._eigenvalues as myEigenvaluePackage
import numpy as np

print(myEigenvaluePackage.eigenvalues(np.array([[1, 0], [0, -1]])))
```

### CODE REVIEW

File: src/generalizedSelfAdjointEigenSolver.cpp [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#include <Eigen/Dense>
+#include <Eigen/Eigenvalues> 
+#include <pybind11/eigen.h>
+#include <pybind11/pybind11.h>
+
+namespace py = pybind11;
+
+Eigen::SelfAdjointEigenSolver<Eigen::MatrixXd> thisSolver;
```
Comment (akhelil) [inline src/generalizedSelfAdjointEigenSolver.cpp:8]: Potential typo here (LINE 8): 

[X] Eigen::SelfAdjointEigenSolver<Eigen::MatrixXd> thisSolver;
--> Eigen::GeneralizedSelfAdjointEigenSolver<Eigen::MatrixXd> thisSolver;

File: src/generalizedSelfAdjointEigenSolver.cpp
Comment (akhelil) [inline src/generalizedSelfAdjointEigenSolver.cpp:1]: This solves the Self Adjoint Eigen Problem, NOT the _Generalized_ Self Adjoint Eigen Problem

File: CMakeLists.txt [hunk 2]
```diff
@@ -14,5 +14,10 @@ FetchContent_Declare(
 FetchContent_MakeAvailable(eigen)
 
 pybind11_add_module(_core MODULE src/main.cpp)
+pybind11_add_module(_eigenvalues MODULE src/generalizedSelfAdjointEigenSolver.cpp)
```
Comment (akhelil) [inline CMakeLists.txt]: Remove "generalized"

**Comment by skkwan** [comment]
Hi @Nanoemc @luclepot , this branch has been merged so the `.eigenvalue_` library can be tested with the following minimal example at least:

> This PR adds a function to get the eigenvalues of a matrix. For example:
> 
> ```python
> import tac_hep_coding_jam._eigenvalues as myEigenvaluePackage
> import numpy as np
> 
> print(myEigenvaluePackage.eigenvalues(np.array([[1, 0], [0, -1]])))
> ```

**Pull request merged**